server/src/main/java/keywhiz/service/permissions/AutomationClientDelegatingPermissionCheck.java

```diff
@@ -0,0 +1,76 @@
+package keywhiz.service.permissions;
+
+import com.codahale.metrics.MetricRegistry;
+import com.google.inject.Inject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AutomationClientDelegatingPermissionCheck implements PermissionCheck{
+  private static final Logger logger = LoggerFactory.getLogger(
+      AutomationClientDelegatingPermissionCheck.class);
+
+  private PermissionCheck delegate;
+  private MetricRegistry metricRegistry;
+
+  @Inject
+  public AutomationClientDelegatingPermissionCheck(MetricRegistry metricRegistry, PermissionCheck delegate) {
+    this.delegate = delegate;
+    this.metricRegistry = metricRegistry;
+  }
+
+  public boolean isAllowed(KeywhizPrincipal source, String action, Object target) {
+    boolean hasPermission;
+
+    if (isAutomation(source)) {
+      hasPermission = true;
+    } else {
+      hasPermission = delegate.isAllowed(source, action, target);
+    }
+
+    int hasPermissionSuccessMetricInt = hasPermission ? 1 : 0;
+    String successMetricName = MetricRegistry.name(AutomationClientDelegatingPermissionCheck.class, "isAllowed", "success", "histogram");
+    metricRegistry.histogram(successMetricName).update(hasPermissionSuccessMetricInt);
+
+    int hasPermissionFailureMetricInt = hasPermission ? 0 : 1;
+    String failureMetricName = MetricRegistry.name(AutomationClientDelegatingPermissionCheck.class, "isAllowed", "failure", "histogram");
+    metricRegistry.histogram(failureMetricName).update(hasPermissionFailureMetricInt);
```

Comment:
Just to shorten things up a bit, you could move all the metrics stuff into an emitMetrics() helper method. That's actually one of the really nice things about the built-in metrics annotations is that they really hide all of this boilerplate and allow the method to just focus on the business logic.